icu4c/source/i18n/plurrule.cpp

```diff
@@ -1533,20 +1530,38 @@ FixedDecimal::FixedDecimal() {
 
 FixedDecimal::FixedDecimal(const UnicodeString &num, UErrorCode &status) {
     CharString cs;
-    cs.appendInvariantChars(num, status);
+    int32_t parsedExponent = 0;
+
+    int32_t exponentIdx = num.indexOf(LOW_E);
+    if (exponentIdx < 0) {
+        exponentIdx = num.indexOf((UChar) 0x0045); // capital E
```

Comment:
`indexOf(u'E')`

Since we adopted C++11 we can use UTF-16 string and character literals.

Reply:
Done.